app/controllers/YourBillController.scala

```diff
@@ -51,13 +51,21 @@ class YourBillController @Inject() (
     }
   }
 
-  private def displayPage(journey: Journey.AfterEligibilityChecked)(implicit request: Request[_]): Result =
-    Ok(
-      views.yourBillIs(
-        YourBillController.overDuePayments(journey.eligibilityCheckResult),
-        journey.taxRegime
+  private def displayPage(journey: Journey.AfterEligibilityChecked)(implicit request: Request[_]): Result = {
+    try {
+      Ok(
+        views.yourBillIs(
+          YourBillController.overDuePayments(journey.eligibilityCheckResult),
+          journey.taxRegime
+        )
       )
-    )
+    } catch {
+      // SA only: if MainTrans is not found, see README
+      case e: NoSuchElementException =>
```

Comment:
it might be safer to use and easier to trace if we had a specific exception for this just in case something is added in the future which can also give a `NoSuchElementException`, we could have something like this for example:
```scala
object MainTrans {
  case object UnknownMainTrans extends Throwable
}

....
// where we're getting from the ChargeTypeMessages map
val message = ChargeTypeMessages.chargeFromMTrans.getOrElse(
  mTrans,
  throw new UnknownMainTrans
)
```